### ISSUE TITLE
Question: How to efficiently get attention weights with beam search decoding?

### ISSUE THREAD

**Issue by ItakeLs** [linto-ai/whisper-timestamped#39]
Hello, just wanted to ask, if inference has to happen twice during beam search decoding, I'm trying to understand if it is a limitation of the whisper architecture or something that is very difficult to implement in one pass, to see if it is an area I can help in.

What would happen if you got the attention while it was decoding on the fly for the naive solution?

**Comment by Jeronymous** [comment]
I'm sorry, I'm not understanding the question. Can you rephrase it?

What I can say is that, currently in whisper-timestamped, getting attention weights on the fly is only possible when the decoding is greedy (no beam search).
It would be possible to get them with beam search only by forking whisper and implementing that in another fashion.

**Comment by ItakeLs** [comment]
Sorry for the bad wording, but you answered my question perfectly. I was wondering if it was a limit of the whisper architecture (some reason why you couldn't use the attention weights), but if I understood you correctly, it turns out that due to the way the whisper code was implemented the attention weights cannot be obtained during beam search.

So essentially, if whisper was modified to return the attention weights for each token in a segment, you could do it in one pass?

**Comment by Jeronymous** [comment]
> So essentially, if whisper was modified to return the attention weights for each token in a segment along with the rest of the segment details, you could do it in one pass?

yes :+1:
